android/app/src/main/java/com/microsoft/codepush/react/CodePushDialog.java

```diff
@@ -11,30 +11,23 @@
 
 public class CodePushDialog extends ReactContextBaseJavaModule{
 
-    private Activity mainActivity;
+    private CodePushNativeModule mCodePushNativeModule;
 
-    public CodePushDialog(ReactApplicationContext reactContext, Activity mainActivity) {
+    public CodePushDialog(ReactApplicationContext reactContext, CodePushNativeModule codePushNativeModule) {
         super(reactContext);
-        this.mainActivity = mainActivity;
+        mCodePushNativeModule = codePushNativeModule;
     }
 
     @ReactMethod
     public void showDialog(String title, String message, String button1Text, String button2Text,
                       final Callback successCallback, Callback errorCallback) {
-        AlertDialog.Builder builder = new AlertDialog.Builder(mainActivity);
+        AlertDialog.Builder builder = new AlertDialog.Builder(mCodePushNativeModule.getCurrentActivityPublic());
 
```

Comment:
Since this is a native module as well, can't you just call getCurrentActivity to get the Activity?